pkg/monitor/agent.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitor
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+
+	agentquerier "github.com/vmware-tanzu/antrea/pkg/agent/querier"
+	"github.com/vmware-tanzu/antrea/pkg/apis/clusterinformation/v1beta1"
+	clientset "github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned"
+	"github.com/vmware-tanzu/antrea/pkg/querier"
+)
+
+type agentMonitor struct {
+	client  clientset.Interface
+	querier agentquerier.AgentQuerier
+}
+
+// NewAgentMonitor creates a new agent monitor.
+func NewAgentMonitor(
+	client clientset.Interface,
+	querier agentquerier.AgentQuerier,
+) *agentMonitor {
+	return &agentMonitor{client: client, querier: querier}
+}
+
+// Run creates AntreaAgentInfo CRD first after controller is running.
+// Then updates AntreaAgentInfo CRD every 60 seconds.
+func (monitor *agentMonitor) Run(stopCh <-chan struct{}) {
+	klog.Info("Starting Antrea Agent Monitor")
+	agentCRD := monitor.getAgentCRD()
+	var err error = nil
+
+	// Initialize agent monitoring CRD.
+	if agentCRD == nil {
+		agentCRD, err = monitor.createAgentCRD()
+		if err != nil {
+			klog.Errorf("Failed to create agent monitoring CRD %v : %v", agentCRD, err)
```

Comment:
I unified the log message style, this is reverting it. So do several places.

Reply:
Updated according to https://github.com/vmware-tanzu/antrea/pull/533. All of they are in agent.Run()